### ISSUE TITLE
1.9.1 fixes minikube --vm-driver=none partially and breaks old behavior with delete

### ISSUE THREAD

**Issue by matti** [GoogleContainerTools/skaffold#4131]
The 1.9.1 release notes don't mention anything about https://github.com/GoogleContainerTools/skaffold/commit/377da24780f44f78547cefbb78535c58fff7f426 which was also released with this release ping @nkubala and @dgageot 

In 1.8.0 the `skaffold delete` gave a warning, but continued with the delete:
```
$ skaffold delete
WARN[0000] Could not get minikube docker env, falling back to local docker daemon: getting minikube env: running [minikube docker-env --shell none]
 - stdout: 
 - stderr: "* 'none' driver does not support 'minikube docker-env' command\n": exit status 64 
Cleaning up...
(removed cleaning up output)
```

With 1.9.1 the command exits and does not do the cleanup:
```
$ skaffold delete
creating runner: creating builder: getting docker client: getting minikube env: running [minikube docker-env --shell none]
 - stdout:                                                                                                                                                                                   
 - stderr: "* 'none' driver does not support 'minikube docker-env' command\n": exit status 64 
(the command exits here)
$ 
```

Current workaround: stay with 1.8.0 with `minikube --vm-driver=none`

**Comment by dgageot** [comment]
That’s really strange... I must have missed something. Can you share a log in debug mode with the old and the new versions? That would be super useful

**Comment by matti** [comment]
sure, how do I enable debug mode?

**Comment by dgageot** [comment]
Skaffold dev -vdebug

**Comment by dgageot** [comment]
@matti I can't reproduce. The output logs would be really helful

**Comment by matti** [comment]
with 1.8.0:
```
...
DEBU[0000] validating yamltags of struct BuildType      
DEBU[0000] validating yamltags of struct LocalBuild     
DEBU[0000] validating yamltags of struct DeployConfig   
DEBU[0000] validating yamltags of struct DeployType     
DEBU[0000] validating yamltags of struct KustomizeDeploy 
DEBU[0000] validating yamltags of struct KubectlFlags   
INFO[0000] Using kubectl context: minikube              
DEBU[0000] Using builder: local                         
DEBU[0000] Running command: [minikube docker-env --shell none] 
creating runner: creating builder: getting docker client: getting minikube env: running [minikube docker-env --shell none]
 - stdout:                                                                                                                                                                                   
 - stderr: "* 'none' driver does not support 'minikube docker-env' command\n": exit status 64 
(process exits)
```

with 1.9.1:
```
INFO[0000] Using kubectl context: minikube              
DEBU[0000] Using builder: local                         
DEBU[0000] Running command: [minikube docker-env --shell none] 
WARN[0000] Could not get minikube docker env, falling back to local docker daemon: getting minikube env: running [minikube docker-env --shell none]
 - stdout: 
 - stderr: "* 'none' driver does not support 'minikube docker-env' command\n": exit status 64 
DEBU[0000] setting Docker user agent to skaffold-v1.8.0 
Cleaning up...
DEBU[0000] Running command: [kustomize build kustomize/envs/superbot-club] 
DEBU[0000] Command output: [apiVersion: v1
kind: ServiceAccount
```

it continues

**Comment by dgageot** [comment]
I'm not following you. The fact that It continues with 1.9.1 (the latest) is a good thing, right? It's also what I do see when I test it myself.

**Comment by matti** [comment]
@dgageot sorry, I was in rush and pasted the wrong snippets - Updated the comment above. With 1.8.0 it continues, with 1.9.1 it exits.

**Comment by matti** [comment]
and I (or my co-worker) was able to repro this today again just by changing the skaffold version back to 1.8.0 (thanks to asdf-vm.com)

**Comment by dgageot** [comment]
I'm going to close that issue because the fix was not released in v1.9.1. However, it is in master branch. v1.9.1 was the cherry-picking of a single PR, not the one that fixes this.
